xprof/xprof.rb.in

```diff
@@ -370,132 +420,151 @@ def enable_events_cl(lttng_session_uuid, channel_name, tracing_mode: 'default',
+         end
+
+  if opts && OPTIONS[:analysis]
+    opts << "--output #{thapi_trace_dir_tmp}"
+    exec("#{BINDIR}/babeltrace_thapi #{opts} -- #{lttng_trace_dir_tmp}")
+    FileUtils.rm_f(lttng_trace_dir_tmp)
+  else
+    FileUtils.mkdir_p(File.dirname(thapi_trace_dir_tmp))
+    FileUtils.mv(lttng_trace_dir_tmp, thapi_trace_dir_tmp)
+  end
+
+  # Global Barrier
+  global_barrier(global_barrier_h)
+
+  # Replace mpi_job_id with a better name
+  return unless mpi_master?
+
+  thapi_trace_dir_tmp_root = File.dirname(thapi_trace_dir_tmp)
+  FileUtils.mv(thapi_trace_dir_tmp_root, thapi_trace_dir_root)
+  thapi_trace_dir_root
+end
+
+# Start, Stop lttng, amd do the on-node analsysis
+def trace_and_on_node_processing(usr_argv)
   # All masters setup a future global barrier
   global_barrier_h = init_global_barrier if mpi_local_master?
 
   # Load Tracers and APILoaders Lib
   backends, h = env_tracers
 
-  # All Need to set the LLTTNG_HOME
+  # All ranks need to set the LLTTNG_HOME env
   # so they can have access to the daemon
-  lttng_output, lttng_session_uuid, lttng_home = get_lttng_paths
-  ENV['LTTNG_HOME'] = lttng_home
+  ENV['LTTNG_HOME'] = lttng_home_dir
   # Only local master spawn LTTNG daemon and start session
-  setup_lttng(lttng_output, lttng_session_uuid, lttng_home, backends) if mpi_local_master?
+  setup_lttng(backends) if mpi_local_master?
   local_barier('waiting_for_lttng_setup')
   # Launch User Command
   launch_usr_bin(h, usr_argv)
 
   # We need to be sure that all the local ranks finished
   # before local master stops the lttng session
   local_barier('waiting_for_application_ending')
-  teardown_lttng(lttng_session_uuid, lttng_home) if mpi_local_master?
-
-  # Ensure that all traces have been written so we can post-process them
-  global_barrier(global_barrier_h) if mpi_local_master?
-
-  # Master will move all the folders to a better folder
-  rename_to_human_readable_folder(File.dirname(lttng_output)) if mpi_master?
-end
-
-#    _                                   _
-#   |_)  _. |_   _  | _|_ ._ _.  _  _     )
-#   |_) (_| |_) (/_ |  |_ | (_| (_ (/_   /_
-#
+  return unless mpi_local_master?
 
-def last_trace_saved
-  Dir[File.join(env_fetch_first('HOME'), 'lttng-traces', 'thapi--*')].max_by { |f| File.mtime(f) }
+  teardown_lttng
+  # Preprocess trace
+  on_node_processing(backends, global_barrier_h)
 end
 
-# TODO: Use babeltrace_thapi as a LIB not a binary
-def postprocess(folder, _options)
-  return unless mpi_master?
+def global_processing(folder)
+  raise unless mpi_master?
 
   LOGGER.info { "postprocess #{folder}" }
-  puts("Trace Location #{folder}")
   backends = OPTIONS[:'backend-name'].join(',')
   backend_level = OPTIONS[:backend].filter { |nl| nl.include?(':') }.join(',')
 
   if OPTIONS.include?(:trace)
-    opts = 'trace'
+    if folder.include?('_interval') || folder.include?('_aggreg')
+      raise 'Cannot show the trace of an interval or aggregated trace'
+    end
+
+    opts = 'trace '
```

Comment:
This fix `trace` not working...  :( So we should merge ASAP, or can do a smaller PR with only those `space` fix.